src/filters/WorldFilter.cpp

```diff
@@ -0,0 +1,129 @@
+//
+// Created by kjhertenberg on 13-5-19.
+//
+
+#include "roboteam_proto/messages_robocup_ssl_detection.pb.h"
+#include <filters/WorldFilter.h>
+
+namespace world {
+
+    WorldFilter::WorldFilter() {
+        blueBots.clear();
+        yellowBots.clear();
+        balls.clear();
+    }
+
+// if we get a new frame we update our observations
+    void WorldFilter::addFrame(const proto::SSL_DetectionFrame &msg) {
+        const double filterGrabDistance = 0.5;
```

Comment:
Why 0.5? Should be able to be passed to function

Reply:
I would not pass a parameter that is constant to a function unless there's a need to change this.

This distance can be tuned further ofcourse. Ideally it's as low as possible but obviously our filters can be wrong at times so there needs to be some leeway for mistakes.

Reply:
You can ofcourse make it a constant class variable